onnx2torch/node_converters/eye.py

```diff
@@ -0,0 +1,50 @@
+__all__ = [
+    'OnnxEyeLike',
+]
+
+import torch
+from torch import nn
+
+from onnx2torch.node_converters.registry import add_converter
+from onnx2torch.onnx_graph import OnnxGraph
+from onnx2torch.onnx_node import OnnxNode
+from onnx2torch.utils.common import OnnxToTorchModule
+from onnx2torch.utils.common import OperationConverterResult
+from onnx2torch.utils.common import onnx_mapping_from_node
+
+
+class OnnxEyeLike(nn.Module, OnnxToTorchModule):  # pylint: disable=missing-docstring
+    def __init__(self, eyelike_k: int, dtype: torch.dtype = None):
+        super().__init__()
+        self.dtype = dtype
+        self.eyelike_k = eyelike_k
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:  # pylint: disable=missing-function-docstring
+        if len(x.shape) != 2:
+            raise ValueError(f'EyeLike only supports 2D tensors, got {len(x.shape)}')
+
+        size_n = x.size(dim=0)
+        size_m = x.size(dim=1)
+        if self.eyelike_k > size_n:
+            raise ValueError(
+                f'the eyelike_k value is {self.eyelike_k},'
+                f'but x shape is {(size_n, size_m)}, eyelike_k should be greater than {size_n}'
+            )
+
+        if self.eyelike_k == 0:
+            return torch.eye(n=size_n, m=size_m, dtype=self.dtype)
+
+        k_tensor = torch.zeros(size_n, self.eyelike_k, dtype=self.dtype)
+        eye_tensor = torch.eye(n=size_n, m=size_m - self.eyelike_k, dtype=self.dtype)
+        return torch.concat([k_tensor, eye_tensor], axis=1)
+
+
+@add_converter(operation_type='EyeLike', version=9)
+def _(node: OnnxNode, graph: OnnxGraph) -> OperationConverterResult:  # pylint: disable=unused-argument
+    node_attributes = node.attributes
+    eyelike_k = node_attributes.get('k', 0)
+    dtype = node_attributes.get('dtype', torch.float32)
```

Comment:
From EyeLike opset documentation:

> dtype : int
>     (Optional) The data type for the elements of the output tensor. If not specified,the data type of the input tensor T1 is used. If input tensor T1 is also notspecified, then type defaults to 'float'.


So, we have the following logic:

1. If `dtype` is not present, then dtype of ouput tensor should be taken from input tensor (x).
2. If `dtype` is present, then dtype of output tensor should be specified `dtype`, but mapped from ONNX dtype to PyTorch analog (leave it to me, don't implement).

> If input tensor T1 is also notspecified, then type defaults to 'float'.

I think it is not possible in pytorch to create tensor without a type, we can skip this option.